diff --git a/lib/notify.c b/lib/notify.c
index 64ecc619..bf6020d1 100644
--- a/lib/notify.c
+++ b/lib/notify.c
@@ -79,12 +79,6 @@ set_script_env(uid_t uid, gid_t gid)
 		}
 	}
 
-	/* Clear any extra supplementary groups */
-	if (setgroups(1, &gid) < 0) {
-		log_message(LOG_ALERT, "Couldn't setgroups: %u (%m)", gid);
-		return true;
-	}
-
 	if (uid != our_uid) {
 		if (setuid(uid) < 0) {
 			log_message(LOG_ALERT, "Couldn't setuid: %u (%m)", uid);
@@ -511,13 +505,6 @@ find_path(notify_script_t *script)
 		sgid_list = MALLOC(((size_t)sgid_num + 1) * sizeof(gid_t));
 		sgid_num = getgroups(sgid_num, sgid_list);
 		sgid_list[sgid_num++] = our_gid;
-
-		/* Clear the supplementary group list */
-		if (setgroups(1, &script->gid)) {
-			log_message(LOG_INFO, "Unable to set supplementary gids (%m)");
-			ret_val = EACCES;
-			goto exit;
-		}
 	}
 
 	for (p = path; ; p = subp)
@@ -605,13 +592,6 @@ exit:
 	if (script->uid != our_uid && seteuid(our_uid))
 		log_message(LOG_INFO, "Unable to restore euid after script search (%m)");
 
-	/* restore supplementary groups */
-	if (sgid_list) {
-		if (setgroups((size_t)sgid_num, sgid_list))
-			log_message(LOG_INFO, "Unable to restore supplementary groups after script search (%m)");
-		FREE(sgid_list);
-	}
-
 exit1:
 	/* We tried every element and none of them worked. */
 	if (got_eacces) {
